gcc-cci/Dockerfile

```diff
@@ -0,0 +1,116 @@
+FROM ubuntu:xenial as base
+
+LABEL maintainer="Conan.io <info@conan.io>"
+
+ARG COMPILER_VERSION
+
+ENV PYENV_ROOT=/opt/pyenv \
+    PATH=/opt/pyenv/shims:/usr/local/bin:${PATH} \
+    CXX=/usr/bin/g++ \
+    CC=/usr/bin/gcc \
+    PYTHON_VERSION=3.8.2 \
+    CMAKE_VERSION=3.17.2 \
+    GCC_VERSION=$COMPILER_VERSION
+
+RUN apt-get -qq update \
+    && apt-get -qq install -y --no-install-recommends \
+       sudo \
+       build-essential \
+       wget \
+       git \
+       libc6-dev \
+       gcc \
+       libgmp-dev \
+       libmpfr-dev \
+       libmpc-dev \
+       nasm \
+       dh-autoreconf \
+       libffi-dev \
+       libssl-dev \
+       pkg-config \
+       subversion \
+       zlib1g-dev \
+       libbz2-dev \
+       libsqlite3-dev \
+       libreadline-dev \
+       xz-utils \
+       curl \
+       libncurses5-dev \
+       libncursesw5-dev \
+       liblzma-dev \
```

Comment:
Can the `dev`packages and other utils be removed at the end of the docker script?
Keeping only the runtime libraries reduces size and avoids accidental dependencies on system libraries.
I assume keeping only the regular packages will keep everything working?

cci recipes should use the packages provided by cci.
e.g. `libgmp`, `libmpfr`, `libmpc`, `nasm`, `autoconf`, `zlib`, 'libbz2', `sqlite3`, `ncurses`, `flex`, `xz-utils`, ...

Reply:
Some of them, yes. We need to filter it, but it's an excellent idea.

Reply:
Indeed. `libc6-dev` should be kept installed, as probably `pkg-config`, `git`, `subversion`...

Reply:
I know some packages were installed to build python, libsqlite3-dev for instance, so you are right. I'll comb through the packages and check what can removed. As the images have unit tests, it's possible to see what breaks after removing a system package.

Reply:
What makes this hard is that e.g. `libsqlite3-dev` should be removed, but `sqlite3` should be kept installed.

Reply:
But if you install everything into `usr/local` won't it make sense to start "final" image from scratch and just copy this whole directory into the new one? So whatever is needed to build gcc stays in the "build" image?